src/plugins/share/public/plugin.ts

```diff
@@ -74,14 +84,17 @@ export class SharePlugin
     >
 {
   private config: ClientConfigType;
-  private readonly shareMenuRegistry = new ShareMenuRegistry();
+  private readonly shareMenuRegistry?: ShareMenuRegistry;
   private readonly shareContextMenu = new ShareMenuManager();
   private redirectManager?: RedirectManager;
   private url?: BrowserUrlService;
   private anonymousAccessServiceProvider?: () => AnonymousAccessServiceContract;
+  private kibanaVersion: string;
 
   constructor(private readonly initializerContext: PluginInitializerContext) {
     this.config = initializerContext.config.get<ClientConfigType>();
+    this.kibanaVersion = initializerContext.env.packageInfo.version;
+    this.shareMenuRegistry = new ShareMenuRegistry(this.config.new_version.enabled);
```

Comment:
Let's used named parameters for this constructor
```suggestion
    this.shareMenuRegistry = new ShareMenuRegistry({ newVersionEnabled: this.config.new_version.enabled });
```